packages/trie/src/trie.ts

```diff
@@ -113,6 +113,50 @@ export class Trie {
     return new Trie(opts)
   }
 
+  static async fromProof(proof: Proof, opts?: TrieOpts): Promise<Trie> {
+    const trie = await Trie.create(opts)
+    if (opts?.root && !equalsBytes(opts.root, trie.hash(proof[0]))) {
+      throw new Error('Invalid proof provided')
+    }
+    const opStack = proof.map((nodeValue) => {
+      return {
+        type: 'put',
+        key: Uint8Array.from(trie.hash(nodeValue)),
+        value: nodeValue,
+      } as PutBatch
+    })
+    await trie._db.batch(opStack)
+    await trie.persistRoot()
+    return trie
+  }
+
+  /**
+   * Verifies a proof.
+   * @param rootHash
+   * @param key
+   * @param proof
+   * @param opts Trie options
+   * @throws If proof is found to be invalid.
+   * @returns The value from the key, or null if valid proof of non-existence.
+   */
+  static async verifyProof(
+    rootHash: Uint8Array,
+    key: Uint8Array,
+    proof: Proof,
+    opts?: TrieOpts
```

Comment:
Three, actually, because the first node in the proof is also the root node.  I didn't write the method, I just made it static, and left it mostly the way it was.  The top `rootHash` parameter is definitely not needed, and the "opts" here really just needs to set the `keyHashing` behavior.